File: app/src/main/java/org/test/kotlin/demo/ApplicationInjector.java
package org.test.kotlin.demo;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {ApplicationModule.class, AndroidSupportInjectionModule.class})
interface ApplicationInjector extends AndroidInjector<Application> {
}

File: app/src/main/java/org/test/kotlin/demo/ApplicationInjector.kt
package org.test.kotlin.demo

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class, AndroidSupportInjectionModule::class])
internal interface ApplicationInjector : AndroidInjector<Application>
